server/middleware/form/__tests__/process.spec.js

```diff
@@ -475,10 +475,7 @@ describe('Process middleware', () => {
 
         processMiddleware(req, res, next);
         expect(req.form).toBeDefined();
-        expect(req.form.data).toBeDefined();
-        expect(req.form.data['test-field-1']).toEqual('-01-01');
-        expect(req.form.data['test-field-2']).toEqual('2021-01-01');
-        expect(req.form.data['test-field-3']).toEqual('2000-01-01');
+        expect(req.form.data).toBeUndefined();
```

Comment:
This is undefined because next is being called with a validation error before data is set. Since validation is tested in [validation.spec.js](https://github.com/UKHomeOffice/hocs-frontend/blob/hocs-6656_invalidDateFix/server/middleware/form/__tests__/validation.spec.js) we should make sure there are tests there for this change and tests that cover `validateForm` throwing a validation error. Then if there are no valid dates that will cause this path to be followed this test can be removed. Otherwise update the dates to properly test the changed functionality.

Reply:
Updated with valid dates for this test.